src/base/rss/rss_feed.cpp

```diff
@@ -280,9 +280,10 @@ void Feed::load()
 {
     QMetaObject::invokeMethod(m_serializer
             , [serializer = m_serializer, url = m_url
-                , path = (m_session->dataFileStorage()->storageDir() / m_dataFileName)]
+            , path = (m_session->dataFileStorage()->storageDir() / m_dataFileName)
+            , fileMaxSize = (10 * 1024 * 1024) + 5 * 1024 * m_session->maxArticlesPerFeed()]
```

Comment:
since it will depend on others, I'll remove the base size and give more margin to each article (if possible):
```suggestion
            , fileMaxSize = m_session->maxArticlesPerFeed() * 10 * 1024]
```

Reply:
Personally, it seems to me extremely illogical when a program cannot load a file previously created by itself.

Reply:
>Personally, it seems to me extremely illogical when a program cannot load a file previously created by itself.

It is not impossible if you think about it. It depends on how you write/load a file. If streaming then file size is not an problem. If the whole file is read into memory then (obviously) you are bounded by memory size.

Reply:
> It is not impossible if you think about it. It depends on how you write/load a file. If streaming then file size is not a problem. If the whole file is read into memory then (obviously) you are bounded by memory size.

Memory size is a natural limitation. If the user cannot load the file because the resources of his system do not allow it, then this is quite normal. But in this case (speaking of a symmetrical way of creating/loading a file), the user is unlikely to be able to create such a file due to the same limitations of his system resources.

Reply:
>If the user cannot load the file because the resources of his system do not allow it, then this is quite normal. 

Off topic. I consider the app crashing/failing is not a good outcome of out-of-memory. Rather, I would expect only the specific operation to fail and not the whole app.

>But in this case (speaking of a symmetrical way of creating/loading a file), the user is unlikely to be able to create such a file due to the same limitations of his system resources.

Off topic. If the write file and read file occurs closely then it make sense. But if the reading occurs later then IMO the system free resources could be undetermined (considering an interactive desktop env).

Reply:
> Off topic. I consider the app crashing/failing is not a good outcome of out-of-memory. Rather, I would expect only the specific operation to fail/stop and not the whole app.

These are interesting topics for reflection and discussion. But I still think that you are more likely to ruin the UX than to improve it by introducing such "hardcoded" limits.